### PR TITLE
Axi4: Add Axi4IdRemover and tests

### PR DESCRIPTION
**Pull request by kleinai**
Adds a component to perform ID reflection to Axi buses without ID support. Integral part of the translation to Axi4Lite.

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4IdRemover.scala [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  } else {
+    val txnActive = RegInit(False)
+    txnActive setWhen(io.input.ar.fire)
+    val txnId = RegNextWhen(io.input.ar.id, io.input.ar.fire)
+
+    io.output.ar << io.input.ar.continueWhen(!txnActive)
+
+    txnActive clearWhen(io.output.r.last && io.output.r.fire)
+
+    io.input.r.arbitrationFrom(io.output.r.continueWhen(txnActive))
```
Comment (Dolu1990) [inline lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4IdRemover.scala:104]: I'm not sure why continueWhen here. I mean, r stream should always come at least one cycle after ar right ? (AXI spec)

Comment (kleinai) [reply lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4IdRemover.scala:104]: Yeah I'm not sure why I put that in there. I'm still playing around with this for Axi to Lite conversion.

**Comment by Dolu1990** [comment]
@kleinai What is the status ^^ ?

**Comment by kleinai** [comment]
I believe it's done. It has tests and I rebased it onto dev without issue.

**Comment by Dolu1990** [comment]
thanks ^^

**Pull request merged**